### DISCUSSION TITLE
Calculating ONDJFM mean for each season

### DISCUSSION THREAD

**Discussion by dkumarjnu** [pydata/xarray#6865] [Q&A] [answered]
Hi Folks, 

I was looking around for the solution but could not find a concrete one. 

From the daily data I am trying to select the ONDJFM months in a way that, the contribution to each season comes as 3 months each from two consecutive years. This means, for one season the data shall start at 1st October and spans till 31st March next year (e.g. 1901-10-01 to 1902-03-31 as one season). 

So far, I was able to create the slice as following:
```python
ONDJFM = f.time.dt.month.isin([10,11,12, 1, 2, 3])

oct_mar=f[ONDJFM].sel(time=slice('1901-10', '2021-03'))

oct_mar_resample = oct_mar.resample(time='182D')
```
gives
```
DataArrayResample, grouped over '__resample_dim__'
239 groups with labels 1901-10-01, ..., 2020-11-03.
```

Is it possible to create groups of each season (roughly 182 days) and calculate a mean?
The data also contains the leap days. I have tried resampling it over 182 days but it gives me 239 groups instead of 121 years. 

I have also tried removing the leap dates and tried resampling but not getting the desired result. Any lead is appreciated. 

Cheers, 
Dhirendra

**Comment by dcherian** [comment, marked as answer]
On latest Xarray (>2025.04.0), you can do

```python
import xarray as xr
from xarray.groupers import SeasonResampler

ds = xr.tutorial.open_dataset("air_temperature")

ds.resample(time=SeasonResampler(["ONDJFM"])).mean()
```

https://docs.xarray.dev/en/latest/user-guide/time-series.html#handling-seasons
